lib/wings/active_fedora_converter.rb

```diff
@@ -62,26 +64,22 @@ def attributes
     ##
     # @return [ActiveFedora::Base]
     def convert
-      active_fedora_class.new(normal_attributes).tap do |af_object|
-        af_object.id = id unless id.empty?
-        normal_attributes.each_key { |key| af_object.send(:attribute_will_change!, key) }
-        add_access_control_attributes(af_object)
-        convert_members(af_object)
-        convert_member_of_collections(af_object)
-        convert_files(af_object)
+      instance.tap do |af_object|
+        af_object.id ||= id unless id.empty?
+        apply_attributes_to_model(af_object)
       end
     end
 
     def active_fedora_class
-      @active_fedora_class ||=
+      @active_fedora_class ||= # cache the class at the instance level
         begin
           klass = begin
                     resource.internal_resource.constantize
                   rescue NameError
                     Wings::ActiveFedoraClassifier.new(resource.internal_resource).best_model
                   end
 
-          return klass if klass <= ActiveFedora::Base
+          return klass if klass <= ActiveFedora::Common
```

Comment:
we want to catch both `Base` and `File` types. `Common` is a better umbrella. to be honest, we're not seriously testing the interface it defines (i think it would be hard to draw a boundary around it).